### ISSUE TITLE
Cannot use spring-data-mock with entities with composite primary key 

### ISSUE THREAD

**Issue by bfg** [mmnaseri/spring-data-mock#141]
Hi.

I have JPA entity which uses composite primary key:

```java
@Entity
@IdClass(EntityId.class)
public class SomeEntity implements Serializable {
    @Id
    @Column(name = "message_id", updatable = false, nullable = false)
    @Type(type = "pg-uuid")
    private UUID firstId;

    @Id
    @Column(name = "recipient_id", updatable = false, nullable = false)
    @Type(type = "pg-uuid")
    private UUID secondId;
```

Everything works as expected using spring-data repository backed by postgresql, but trying to create spring-data-mock thows an exception:

```
com.mmnaseri.utils.spring.data.error.MultipleIdPropertiesException: There are multiple properties in class SomeEntity that are annotated as the ID property
	at com.mmnaseri.utils.spring.data.domain.impl.id.AnnotatedFieldIdPropertyResolver$1.doWith(AnnotatedFieldIdPropertyResolver.java:37)
```

It would be absolutely awesome if this would be supported by spring-data-mock, because i don't need to bring up the whole spring context for testing.

**Comment by mmnaseri** [comment]
Hi,

Thanks for using this framework. I completely understand your pain, which is what got me to implement this framework in the first place.

**TL;DR** I will try to implement this feature as soon as possible. I need some clarification on the specific use case. Jump to the end of this comment for the question.

** Background **

Currently the `DataStore` abstraction which mimics the behavior of a simple key-value store expects the keys to every entity to be a `Serializable` value. This value is resolved from the entities by accessing a given key property.

** Solution **

The first order of business is to add `EmbeddedKey` as a viable key designator. This will allow for classes marked as `Embeddable` to be used as compound keys. This is without key generation, of course:

````java
@Embeddable
public class AccountOwnershipKey implements Serializable {}

@Entity
public class AccountOwnership {

    @EmbeddedKey
    private AccountOwnershipKey key;

}

public interface AccountOwnershipRepository<AccountOwnership, AccountOwnershipRepository> {
     // omitted for brevity
}

final AccountOwnershipRepository repository  = builder()
                                                 .withoutGeneratingKeys()
                                                 .mock(AccountOwnershipRepository.class);
````

For better support, it would be more extensible to add one level of indirection between keys and data stores, so that there is a `Serializable` class that acts as the intermediary and the key resolver actually figures out the schema for this class and its relationship with the original entity.

** Question **

The big question would be (and this comes from a lack of personal experience on my side) given the following entity definition:

````java
@Entity
public class Membership {

    @Id
    private Long personId;

    @Id
    private Long groupId;

}
````

what would your repository definition look like?

We know that Spring Data requires a single, generic `Serializable` type for the key for each repository. So, it will have to be something like the following:

````java
public interface MembershipRepository extends JpaRepository<Membership, K> {
     // omitted for brevity
}
````

What is `K` in your case? What is your repository definition like?

**Comment by mmnaseri** [comment]
So, started working on this with the following entity class as a sample:

````java
package com.mmnaseri.utils.samples.spring.data.jpa.model;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.IdClass;

/**
 * @author Mohammad Milad Naseri (m.m.naseri@gmail.com)
 * @since 1.0 (7/18/16)
 */
@Entity
@IdClass(AccountAccessKey.class)
public class AccountAccess {

    @Id
    @Column
    private String userId;

    @Id
    @Column
    private String accountId;

    @Column
    private AccountAccessType accessType;

    public String getUserId() {
        return userId;
    }

    public AccountAccess setUserId(String userId) {
        this.userId = userId;
        return this;
    }

    public String getAccountId() {
        return accountId;
    }

    public AccountAccess setAccountId(String accountId) {
        this.accountId = accountId;
        return this;
    }

    public void setAccessType(AccountAccessType accessType) {
        this.accessType = accessType;
    }

    public AccountAccessType getAccessType() {
        return accessType;
    }

}
````

and the following repository definition:

````java
package com.mmnaseri.utils.samples.spring.data.jpa.repository;

import com.mmnaseri.utils.samples.spring.data.jpa.model.AccountAccess;
import com.mmnaseri.utils.samples.spring.data.jpa.model.AccountAccessKey;
import com.mmnaseri.utils.samples.spring.data.jpa.model.AccountAccessType;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

/**
 * @author Mohammad Milad Naseri (m.m.naseri@gmail.com)
 * @since 1.0 (7/18/16)
 */
public interface AccountAccessRepository extends JpaRepository<AccountAccess, AccountAccessKey> {

    List<AccountAccess> findByUserId(String userId);

    List<AccountAccess> findByUserIdAndAccessType(String userId, AccountAccessType accessType);

    List<AccountAccess> findByAccountIdAndAccessType(String accountId, AccountAccessType accessType);

}
````

These are examples taken from the `spring-data-mock-sample-jpa` module which will be made available once the feature is properly supported.

**Comment by JNFrazao** [comment]
Hello,

I am working with Spring Boot tests, using Spring Data Mock and I am facing this issue.

I noticed that your have started working on it, and I would like to know if you have any update regarding it.

Thanks,
Joao
